internal/export/exportfile.go

```diff
@@ -51,43 +52,46 @@ type Signer struct {
 	Signer        crypto.Signer
 }
 
-// MarshalExportFile converts the inputs into an encoded byte array.
-func MarshalExportFile(eb *model.ExportBatch, exposures, revisedExposures []*publishmodel.Exposure, batchNum, batchSize int, signers []*Signer) ([]byte, error) {
+// MarshalExportFile converts the inputs into an encoded byte array and sha256 (base64) of the marshalled protobuf contents.
+func MarshalExportFile(eb *model.ExportBatch, exposures, revisedExposures []*publishmodel.Exposure, batchNum, batchSize int, signers []*Signer) ([]byte, string, error) {
 	// create main exposure key export binary
 	expContents, err := marshalContents(eb, exposures, revisedExposures, int32(batchNum), int32(batchSize), signers)
 	if err != nil {
-		return nil, fmt.Errorf("unable to marshal exposure keys: %w", err)
+		return nil, "", fmt.Errorf("unable to marshal exposure keys: %w", err)
 	}
 
+	digest := sha256.Sum256(expContents)
+	pbSHA := hex.EncodeToString(digest[:])
```

Comment:
Somewhat nit, but it doesn't make sense to calculate this until later when it's needed.